src/Bicep.Core.IntegrationTests/Scenarios/InliningResourcesAndModulesTests.cs

```diff
@@ -512,7 +515,24 @@ public void ObjectVariablesWithUnknownTypeResourceReference_ShouldBeInlined()
                     "Object should be in-lined");
             }
         }
-
+        [TestMethod]
+        public void VariablesParameterResourceReference_ShouldBeInlined()
+        {
+            var result = CompilationHelper.Compile(Services.WithFeatureOverrides(new(ResourceTypedParamsAndOutputsEnabled: true)), 
+            ("main.bicep", @"
+param resourceParameter resource 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30'
+var clientId = resourceParameter.properties.clientId
```

Comment:
Could you also add a negative case here? E.g., `var resourceParameterId = resourceParameter.id`. Then an assertion below should verify that `resourceParameterId` is not inlined.